background.js

```diff
@@ -16,12 +16,17 @@
     }
 
     function storeConfigToLocalStorage(data) {
-        localStorage.setItem(SETUP_STRING, JSON.stringify(data || {}));
+        if (typeof window !== 'undefined') {
```

Comment:
do we need to check undefined here?

Reply:
No need as the bot (which is great) suggested correct changes. 

To answer your question, the issue was that `localStorage` was showing as `undefined`. I believe this happened because the code was accessing the extension page rather than the webpage(`window`). To resolve this, I made changes to the code and replaced `localStorage` with [chrome.storage](url) API ([Reference](https://developer.chrome.com/docs/extensions/reference/api/storage)). 

For now, I’ve tested the updated code thoroughly, and everything works flawlessly. I’ll continue improving it further as needed.